File: cpp/121_Best_Time_To_Buy_And_Sell_Stock/121_Best_Time_To_Buy_And_Sell_Stock.cpp
#include<vector>
#include<algorithm>

using namespace std;

class Solution {
public:
	int maxProfit(vector<int>& prices) {
		if (prices.size() <= 1) return 0;
		int minPrices = prices[0];
		int maxProfit = 0;
		for (auto p : prices)
		{
			if (p < minPrices) minPrices = p;
			else if (maxProfit < (p - minPrices)) maxProfit = p - minPrices;
		}
		return maxProfit;
	}
};

File: python/121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if len(prices)==0:
            return 0
        minPrice = prices[0]
        maxProf = 0
        for p in prices:
            if minPrice>p:
                minPrice = p
            elif maxProf<p-minPrice:
                maxProf = p-minPrice

        return maxProf


import time
start = time.clock()


TestCase = [7,6,5,4]
ExpectOutput = 0


solu = Solution()#先对类初始化，才能进行调用
temp = solu.maxProfit(TestCase)
if (temp == ExpectOutput):
    print('right')
else:
    print('wrong')
    print(temp)

elapsed = (time.clock() - start)
print("Time used:", elapsed)